mongodb/intro/date-aggregation/aggregation-with-date-expression-operators-iii.md

```diff
@@ -0,0 +1,169 @@
+---
+author: Stefan-Stojanovic
+
+levels:
+  - beginner
+  - basic
+  
+aspects:
+  - introduction
+  - new
+
+type: normal
+
+category: how to
+
+---
+
+# Aggregation With $hour, $minutes, $seconds and $milliseconds
+
+---
+## Content
+
+For the document used in aggregation, make sure to check the Previous Document[1] footnote.
+
+### $hour
+
+The `$hour` date operator is used to extract the exact hour the document was created on. The output goes from 0-23.
+
+Example:
+```js
+db.pokedex.aggregate([
+  { $match: { name: "Bulbasaur" } },
+  {
+    $project: {
+      Hour: { $hour: "$Date" },
+      _id: 0
+    }
+  }
+]);
+
+```
+
+Output:
+```js
+{ 
+  "Hour" : 7 
+}
+```
+
+### $minutes
+
+The `$minute` date operator is used to extract the exact minute the document was made on. The output goes from 0 to 59.
+
+Example:
+```js
+db.pokedex.aggregate([
+  { $match: { name: "Bulbasaur" } },
+  {
+    $project: {
+      Minutes: { $minute: "$Date" },
+      _id: 0
+    }
+  }
+]);
+
+```
+Output:
+```js
+{ 
+  "Minutes" : 21 
+}
+```
+
+### $seconds
+
+The `$seconds` date operator is used to extract the exact second the document was made on. Although the output goes from 0 to 59, it is possible for the output to be 60 in case of a leap second.
```

Comment:
Add a footnote for leap second?

Reply:
I can add this link in the `links` section, not sure if I can make a small footnote regarding leap seconds
https://www.mongodb.com/blog/post/mongodb-and-leap-seconds

Reply:
you don't have to explain how mongo prepared for a leap second but rather what a leap second is and why it affects mongodb.

- https://en.wikipedia.org/wiki/Leap_second
- https://www.timeanddate.com/time/leapseconds.html

the main points you should touch are:
- what is a leap second
- why it could affect mongodb
- end with saying that mongo has set systems in place to avoid issues created by leap seconds